gateway/src/shard/impl.rs

```diff
@@ -637,9 +654,18 @@ impl Shard {
         match session.tx.send(message.into_tungstenite()) {
             Ok(()) => {
                 // Tick ratelimiter.
-                session.ratelimit.lock().await.next().await;
-
-                Ok(())
+                let ratelimiter = session.ratelimit.get();
+                if let Some(limiter) = ratelimiter {
+                    limiter.acquire_one().await.map_err(|source| SendError {
```

Comment:
style nit, no biggie 
```suggestion
                if let Some(limiter) = session.ratelimit.get() {
                    limiter.acquire_one().await.map_err(|source| SendError {
```